setup.py

```diff
@@ -25,6 +25,7 @@
     "filelock>=3.0.0,<4.0.0",
     "pip-system-certs>=4.0.0,<5.0.0; platform_system=='Windows'",
     "pyrate-limiter>=2,<3",
+    "more-itertools>=8.0.0,<11.0.0",
```

Comment:
we only need one method that can be replaced by:
```
from itertools import islice

def batcher(iterable, batch_size):
    iterator = iter(iterable)
    while batch := list(islice(iterator, batch_size)):
        yield batch
```
not sure it is necessary to add another lib

Reply:
Ah great!